lib/elixir/lib/module/checker.ex

```diff
@@ -0,0 +1,135 @@
+defmodule Module.Checker do
+  def verify(module_map, _binary) do
+    check_definitions(Enum.reverse(module_map.definitions), module_map)
+    collect_warnings()
+  end
+
+  defp collect_warnings() do
```

Comment:
This is not working correctly because `:elixir_compiler_pid` is not set in the pdict. We need another way to collect the warnings.